packages/firestore/src/model/target_index_matcher.ts

```diff
@@ -51,8 +51,8 @@ import {
 export class TargetIndexMatcher {
   // The collection ID (or collection group) of the query target.
   private readonly collectionId: string;
-  // The single inequality filter of the target (if it exists).
-  private readonly inequalityFilter?: FieldFilter;
+  // The inequality filters of the target (if it exists).
+  private readonly inequalityFilters = new Map<string, FieldFilter>();
```

Comment:
I see.

One issue with the current implementation is that when you have multiple inequalities on the same field, you lose the information for the first one in the map upon inserting the second one in the map (because they have the same key in the map).

so for example the map is
{ "a" : "a <= 10" }
then we insert "a>=1" and the map becomes:
{ "a": "a >= 1" }

(we lost the information about "a<=10".)

That's not good. Although things will continue to work because it seems that the only place these inequality filters are used here is in the `matchesFilter` function. That function only wants to check if the filter operator is an array operator or not. And no matter how you overwrite the "a" entry in this example, the `matchesFilter` function will continue to work the same way (because "a" field cannot be used in an array filter and a non-array filter field).

Let's chat offline tomorrow.

Reply:
Yes, the previous single inequality logic keeps the operator replaced. And the single function that deals with the filter only cares about if it is a 'isArrayOperator' or not. 
``  private matchesFilter(
    filter: FieldFilter | undefined,
    segment: IndexSegment
  ): boolean {
    if (filter === undefined || !filter.field.isEqual(segment.fieldPath)) {
      return false;
    }
    const isArrayOperator =
      filter.op === Operator.ARRAY_CONTAINS ||
      filter.op === Operator.ARRAY_CONTAINS_ANY;
    return (segment.kind === IndexKind.CONTAINS) === isArrayOperator;
  }```

So the map data structure basically follows the same logic, and doesn't care about operator neither.

Reply:
Use the set instead of map per discussion offline.